crates/backend/src/codegen.rs

```diff
@@ -806,6 +806,16 @@ impl TryToTokens for ast::Export {
                     add_check(quote! {
                         let _: #wasm_bindgen::__rt::marker::CheckSupportsConstructor<#class>;
                     });
+
+                    if self.function.r#async {
+                        (quote_spanned! {
+                            self.function.name_span =>
+                            #[deprecated(note = "async constructors are not supported")]
+                            fn constructor() {}
+                            constructor();
```

Comment:
This context is shared with user code, so we need to avoid naming collisions here. I tried this locally and it seems to still work just fine.
```suggestion
                            const _: () = {
                                #[deprecated(note = "async constructors are not supported")]
                                const fn constructor() {}
                                constructor();
                            };
```